app/workers/status_update_worker.rb

```diff
@@ -0,0 +1,38 @@
+class StatusUpdateWorker
+  include Sidekiq::Worker
+
+  attr_accessor :reference, :status
+
+  def perform(params)
+    params.deep_symbolize_keys!
+
+    self.reference = params.fetch(:reference)
```

Comment:
Might be nicer to allow the worker to take `reference` and `status` as specific arguments, and then the caller (in which case the controller) can deal with extracting the values from POST input.